advance-exercises/src/components/01-CompoundComponents.js

```diff
@@ -16,70 +9,125 @@
     - Arrow left, arrow up should select the previous option
 */
 
-import React from 'react';
-import PropTypes from 'prop-types';
+import React from "react";
+import PropTypes from "prop-types";
 
 class RadioGroup extends React.Component {
   static propTypes = {
-    // defaultValue: PropTypes.string,                UN-COMMENT THIS LINE
-    children: PropTypes.shape().isRequired,
+    defaultValue: PropTypes.string,
+    children: PropTypes.shape().isRequired
   };
   render() {
-    return (
-      <div>{this.props.children}</div>
-    );
+    return <div>{this.props.children}</div>;
   }
 }
 
 class RadioOption extends React.Component {
   static propTypes = {
-    // value: PropTypes.string,                       UN-COMMENT THIS LINE
-    children: PropTypes.shape().isRequired,
+    value: PropTypes.string,
+    children: PropTypes.shape().isRequired
+  };
+  state = {
+    isSelected: false
+  };
+  handleClick = radio => {
+    console.log(this.props);
+    this.props.onChange(this.props.value, this.state.isSelected);
+    this.setState({ isSelected: !this.state.isSelected });
   };
-
   render() {
     return (
       <div>
-        <RadioIcon isSelected={false} /> {this.props.children}
+        <RadioIcon
+          isSelected={this.state.isSelected}
+          value={this.props.value}
+          onClick={radio => {
+            this.handleClick(radio);
+          }}
+        />{" "}
+        {this.props.children}
       </div>
     );
   }
 }
 
 class RadioIcon extends React.Component {
+  constructor(props) {
+    super(props);
+    console.log("radio icon", props);
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
   static propTypes = {
-    isSelected: PropTypes.bool.isRequired,
+    isSelected: PropTypes.bool.isRequired
+  };
+  handleOnClick = function() {
+    this.props.onClick();
   };
 
   render() {
     return (
       <div
+        tabIndex="0"
         style={{
-          borderColor: '#ccc',
+          borderColor: "#ccc",
           borderWidth: 3,
-          borderStyle: this.props.isSelected ? 'inset' : 'outset',
+          borderStyle: this.props.isSelected ? "inset" : "outset",
           height: 16,
           width: 16,
-          display: 'inline-block',
-          cursor: 'pointer',
-          background: this.props.isSelected ? 'rgba(0, 0, 0, 0.05)' : '',
+          display: "inline-block",
+          cursor: "pointer",
+          background: this.props.isSelected ? "rgba(0, 0, 0, 0.05)" : ""
         }}
+        onClick={this.handleOnClick}
       />
     );
   }
 }
 
 class CompoundComponents extends React.Component {
   render() {
+    const radio = { am: false, fm: true, tape: false, aux: false };
+    this.handleOnChange = (value, isSelected) => {
```

Comment:
A newline after a variable declaration and before a function start should improve readability.